front-end/src/components/main-page/MainPage.test.js

```diff
@@ -34,16 +37,54 @@ describe('MainPage', () => {
         expect(screen.queryByText("Drag and drop file or upload below.")).not.toBeInTheDocument();
     });
 
-    it('should dispatch openAnalysisPage when View Analysis button is clicked', async() => {
-        store.dispatch(setTranscriptUploadStatus(true));
-        store.dispatch(setProjectIdToBeDisplayed("1"));
+    it('should dispatch setProjectIdToBeDisplayed with correct Project ID if a Project ID is entered', async() => {
+        renderComponent();
+        const dispatch = jest.spyOn(store, 'dispatch');
+        userEvent.type(screen.getByLabelText("Enter Project ID"), "1");
+        expect(dispatch).toHaveBeenCalledWith({ type: 'analyzeTranscript/setProjectIdToBeDisplayed', payload: '1'});
+    });
+
+    it('should enable View Analysis button if a Project ID is entered', async() => {
+        renderComponent();
+        userEvent.type(screen.getByLabelText("Enter Project ID"), "1");
+        expect(screen.getByLabelText("View Analysis")).not.toBeDisabled();
+    });
+
+    it('should disable View Analysis button if no Project ID is entered', async() => {
+        renderComponent();
+        expect(screen.getByLabelText("View Analysis")).toBeDisabled();
+    });
+
+    it('should dispatch openAnalysisPage when valid Project ID is entered and View Analysis button is clicked', async() => {
         store.dispatch(addAnalyzedTranscript({projectId: "1", transcript: [{question: "a", children: {"b": 100,}}]}));
         renderComponent();
         const dispatch = jest.spyOn(store, 'dispatch');
+        userEvent.type(screen.getByLabelText("Enter Project ID"), "1");
         await userEvent.click(screen.getByText('View Analysis'));
         expect(dispatch).toHaveBeenCalledWith({ type: 'switchPage/openAnalysisPage' });
     });
 
+    it('should show an alert and not dispatch openAnalysisPage when invalid Project ID is entered and View Analysis button is clicked', async() => {
+        const mockAlert = jest.spyOn(global, "alert").mockImplementation(); 
+        jest.spyOn(TranscriptAPI, "getAnalysis").mockImplementationOnce(() => {return []});
+        store.dispatch(addAnalyzedTranscript({projectId: "1", transcript: [{question: "a", children: {"b": 100,}}]}));
+        renderComponent();
+        const dispatch = jest.spyOn(store, 'dispatch');
+        userEvent.type(screen.getByLabelText("Enter Project ID"), "2");
+        await userEvent.click(screen.getByText('View Analysis'));
+        expect(mockAlert).toHaveBeenCalledWith("Error in analyzing transcript. Your project ID was not in our database. Please try again.");
+        expect(dispatch).not.toHaveBeenCalledWith({ type: 'switchPage/openAnalysisPage' });
```

Comment:
```suggestion
        expect(dispatch).not.toHaveBeenCalledWith({ type: 'switchPage/openAnalysisPage' });
```

Reply:
Since the 2nd expect implicitly checks the first and the first is refusing to be recognized by jest, the second is good enough